lib/Core/MemoryManager.cpp

```diff
@@ -40,7 +98,31 @@ MemoryObject *MemoryManager::allocate(uint64_t size, bool isLocal,
     klee_warning_once(0, "Large alloc: %lu bytes.  KLEE may run out of memory.",
                       size);
 
+  // Return NULL if size is zero, this is equal to error during allocation
+  if (NullOnZeroMalloc && size == 0)
+    return 0;
+
   uint64_t address = 0;
+  if (DeterministicAllocation) {
+    if ((alignment & -alignment) != alignment) {
+      klee_warning("Only alignment of power of two is supported");
+      return 0;
+    }
+
+    address = ((uint64_t)nextFreeSlot + alignment - 1) & -alignment;
+
+    size_t alloc_size = std::max(size, (uint64_t)1);
+    if (alloc_size > 0 &&
```

Comment:
Given ``alloc_size = std::max(size, (uint64_t) 1);`` then ``alloc_size > 0`` is always true and so should not be part of the conditional.